overlord/devicestate/handlers.go

```diff
@@ -256,6 +256,80 @@ func (m *DeviceManager) doGenerateDeviceKey(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// A registrationContext handles the contextual information needed
+// for the initial registration or a re-registration.
+type registrationContext interface {
+	Device() (*auth.DeviceState, error)
+
+	GadgetForSerialRequestConfig() string
+	SerialRequestExtraHeaders() map[string]interface{}
+	SerialRequestAncillaryAssertions() []asserts.Assertion
+
+	FinishRegistration(serial *asserts.Serial) error
+
+	ForRemodeling() bool
+}
+
+// initialRegistrationContext is a thin wrapper around DeviceManager
+// implementing registrationContext for initial regitration
+type initialRegistrationContext struct {
+	*DeviceManager
```

Comment:
Is this called in the context of DeviceManager anywhere?

Reply:
it's a type, so used?  or am I misunderstanding the question. It is used see here:

https://github.com/snapcore/snapd/pull/6822/files#diff-1b38e3addd0751daf577cc229fdaa4b8R672

calling

https://github.com/snapcore/snapd/pull/6822/files#diff-1b38e3addd0751daf577cc229fdaa4b8R327